### DISCUSSION TITLE
[Q] how can i run model-simplification by Complexity

### DISCUSSION THREAD

**Discussion by copycd** [cnr-isti-vclab/meshlab#1059] [Q&A]
I use a mesh-server and customize the source-code.
I wish to simplify a mesh. but that need to determine target triangle count or percentage.

I want to determine with mesh-Complexity.

ex) QualityThr = 0.5  
      but process runs in target triangle count.
      I with to stop process When complexity is met like "DeciSession.SetTargetSimplices(TargetFaceNum)"

Please~  help me.

ps. 

is that what mean below?
 DeciSession.SetTargetMetric( xx );

**Comment by alemuntoni** [comment]
MeshLab right now supports only simplification with a target triangle number. 
Therefore any other method must be implemented first.
Could you please better explain what you mean exactly with "Complexity"?

**Reply by copycd** [reply]
I am creating service program like meshlab-server.
The program don't know which mesh come.
how many vertex is reduced.
Therefore I need what represents quality parameter.

explain....
ex) A mesh that is composed with 1000 vertex is simple.
      but B mesh that is composed with 1000 vertex is complex.

     input float quality = 0.5f 
     then I wish that A mesh reduce to 200 vertex and B Mesh is 800 vertex.
     input float quality = 0.1f
     then A mesh reduce to 500 vertex and B Mesh is 950 vertex.

is it possible?

**Reply by alemuntoni** [reply]
Nope, as far as I know it is not possible to do this with meshlab.

You did not explain with which metric you define mesh A to be "simple" and mesh B to be "complex", having both meshes the same vertex number. If you first define some sort of metric to define this, then you could write some simple pymeshlab script that runs simplification only when your metric return that a mesh is "complex".

**Reply by copycd** [reply]
thanks for your response.
by any chance.
have i get know if mesh is simple or complex?

**Reply by alemuntoni** [reply]
There is no universal definition of "simple mesh" or "complex mesh". As already said, you first need to define what do you mean with "simple".
